x-pack/plugins/enterprise_search/public/applications/app_search/components/credentials/credentials.test.tsx

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import '../../../__mocks__/kea.mock';
+import '../../../__mocks__/shallow_useeffect.mock';
+
+import React, { useEffect } from 'react';
+import { shallow } from 'enzyme';
+import { useValues, useActions } from 'kea';
+
+import { Credentials } from './credentials';
+import { EuiCopy, EuiPageContentBody } from '@elastic/eui';
+
+import { externalUrl } from '../../../shared/enterprise_search_url';
+
+const getUseEffectUnmountHandler = () => (useEffect as jest.Mock).mock.calls[0][0]();
+
+describe('Credentials', () => {
+  const mockKea = ({ values = {}, actions = {} }) => {
+    const mergedValues = {
+      dataLoading: false,
+      ...values,
+    };
+
+    const mergedActions = {
+      initializeCredentialsData: jest.fn,
+      ...actions,
+    };
+
+    (useValues as jest.Mock).mockImplementationOnce(() => mergedValues);
+    (useActions as jest.Mock).mockImplementationOnce(() => mergedActions);
+  };
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
```

Comment:
I think you could simplify this a lot by taking advantage of the `setMockValues` and `setMockActions` helpers we have in `kea.mock.ts`. You should be able to do something like this:


```suggestion
  // Kea mocks
  const values = {
    dataLoading: false,
  };
  const actions = {
    initializeCredentialsData: jest.fn(),
  };

  beforeEach(() => {
    jest.clearAllMocks();
    setMockValues(values);
    setMockActions(actions);
  });
```

This involves several changes though, so easier just to show a code diff:

https://github.com/JasonStoltz/kibana/compare/JasonStoltz/credentials-view...constancecchen:JasonStoltz/credentials-view

LMK what you think!

Reply:
Oh rad, I didn't realize those existed. Thanks for that.

Reply:
I'm cherry-picking this as well